packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -610,6 +825,66 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   *   Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization
+  */
+  public enableDeploymentAlarms(alarmConfig: DeploymentAlarmConfig) {
+    // Throw an error if alarms array is empty
+    if (alarmConfig.alarms.length === 0) {
+      throw new Error('Specify at least one deployment alarm');
+    }
+    this.deploymentAlarms = {
+      alarmNames: alarmConfig.alarms.map(alarm => alarm.alarmName),
+      enable: true,
+      rollback: alarmConfig.behavior !== AlarmBehavior.FAIL_ON_ALARM,
+    };
+  }
+
+  /**
+   *   Disassociate existing deployment alarms
+  */
+  public disableDeploymentAlarms() {
+    if (this.deploymentAlarms) {
+      this.deploymentAlarms = {
+        alarmNames: [],
+        rollback: false,
+        enable: false,
+      };
+    }
+  }
+
+  /**
+   *   Add an alarm based on ECS metric which appended to the list of deployment alarms at synthesis time
+   *   Customize the alarm threshold and evaluationPeriods using 2nd argument ecsMetricAlarmProps.
+   *   `threshold  value 85`
+   *   `evaluationPeriods value 3`
+  */
+  public createEcsMetricAlarm(metric: EcsMetric, ecsMetricAlarmProps?: EcsMetricAlarmProps): cloudwatch.Alarm {
```

Comment:
[API Change]:

- Make the signature more concise
- Remove the `metric` argument, because this is now contained in the `EcsAlarmProps` interface

```suggestion
  public createAlarm(props: EcsAlarmProps): cloudwatch.Alarm {
```